tests/bluetooth/uuid/src/test_bt_uuid_to_str.c

```diff
@@ -1,6 +1,11 @@
 /* Copyright (c) 2022 Nordic Semiconductor ASA
  * SPDX-License-Identifier: Apache-2.0
  */
+#undef _POSIX_C_SOURCE
+/* Required for strnlen in some C libraries.
+ * Note strnlen is an exception to the coding guidelines rule A.4
+ */
+#define _POSIX_C_SOURCE 200809L
```

Comment:
This proposal enters into the discussion from https://github.com/zephyrproject-rtos/zephyr/pull/67040#discussion_r1462390520 . Given that the current fix as is has no immediate drawbacks, and for the sake or allowing this PR to progress I would rather not re-start that discussion here again.

Reply:
Please make the requested change. Only newlib  and the host libc require these application conformance macros, and we really do not need the documentation repeated here.

Reply:
In general it is not "only newlibc and the host libc" that need it, as much as "any C library which is not minimal or picolibc".
Why do you request this ifdefs be added?

Reply:
@cfriedt Your original proposal has several changes:
a) You use CONFIG_HOST_LIBC but that does not exist I presume if anything that would  need to be EXTERNAL_LIBC
b) Please elaborate on why you want to `#if` these defines and if you will block this PR unless a similar `#if defined(..`  is done
c) Please clarify if you also will block this PR unless the comment is removed.

Reply:
> In general it is not "only newlibc and the host libc" that need it, as much as "any C library which is not minimal or picolibc".

Actually, there are certified ISO C libraries that do not provide the (POSIX) `strnlen()`.

>  Why do you request this ifdefs be added?

Because currently only the host libc and newlib require them. I believe this is self-explanatory. 

> The reason to mention explicitly which functions require this

I understand. However, it's covered in the documentation already. Having a single source of truth is nice (i.e. the documentation).

> is because we are tripping checkpatch

The solution in this PR does trip checkpatch.pl - there are alternatives that are probably more scalable and maintainable that do not trip checkpatch.pl, that do not involve hard-coding values in source files.. but that's a separate topic.

>  Please clarify if you also will block this PR unless the comment is removed.

I would prefer if the comment were removed.

Reply:
>> Why do you request this ifdefs be added?

> Because currently only the host libc and newlib require them. I believe this is self-explanatory.

Unfortunately it is not self explanatory. You are requesting adding extra code to set conditionally a flag which can be set always. And I would prefer to not add unnecessary code.
So I would request you to reconsider rejecting this PR based on that.